src/Workspaces/Core/Portable/Workspace/Workspace.cs

```diff
@@ -1673,6 +1694,28 @@ protected virtual void ApplyCompilationOptionsChanged(ProjectId projectId, Compi
             this.OnCompilationOptionsChanged(projectId, options);
         }
 
+        /// <summary>
+        /// This method is called during <see cref="TryApplyChanges(Solution)"/> to change the default namespace.
+        ///
+        /// Override this method to implement the capability of changing default namespace.
+        /// </summary>
+        protected virtual void ApplyDefaultNamespaceChanged(ProjectId projectId, string? defaultNamespace)
+        {
+            Debug.Assert(CanApplyChange(ApplyChangesKind.ChangeProjectInfo));
+            this.OnDefaultNamespaceChanged(projectId, defaultNamespace);
+        }
+
+        /// <summary>
+        /// This method is called during <see cref="TryApplyChanges(Solution)"/> to change the default namespace.
+        ///
+        /// Override this method to implement the capability of changing a project's name.
+        /// </summary>
+        protected virtual void ApplyProjectNameChanged(ProjectId projectId, string name, string? filePath)
+        {
+            Debug.Assert(CanApplyChange(ApplyChangesKind.ChangeProjectInfo));
```

Comment:
So I'm worried that we have an ApplyChangesKind that covers both the default namespace change above and also the project name here. This PR (currently) isn't updating VisualStudioWorkspace to try making these changes in Visual Studio for Windows. If we did, the challenge is that for us applying a project name change might be difficult to implement (we'd have to try it and see), whereas applying a default namespace change is probably much simpler. I don't exactly like the idea of a different ApplyChangesKind for each type change, but I'm not sure we have an alternative.